.changeset/metal-mails-ring.md

```diff
@@ -0,0 +1,20 @@
+---
+"thirdweb": patch
+---
+
+- Add onClose callback to Connect Details modal
+
+```tsx
+<ConnectButton
+  detailsModal={{
+    onClose: (screen: string) => {
```

Comment:
Rather than burying this in the `detailsModal` prop, could we make it top level?

Reply:
This is only applicable for the Modal and not the whole button, so it contextually makes sense to be nested under "detailsModal".

However it took a while for me to raise the coverage to 80% and I'm willing to do what you said to merge this PR lol

Reply:
Hmm @gregfromstl actually it's cleaner if we keep it inside the `detailsModal` prop. Also by putting onClose at the root level it crashes with the `onClose` of the react native's ConnectButton. (packages/thirdweb/src/react/native/ui/connect/ConnectButton.tsx L114 )

So I'd say we keep things as is. Would you be able to compromise?